File: 2021/10/30/height_checker.py
"""
https://leetcode.com/problems/height-checker/
Author: Andrew Jarombek
Date: 10/30/2021
"""

from typing import List


def height_checker(heights: List[int]) -> int:
    sorted_heights = sorted(heights)
    return sum([1 if heights[i] != sorted_heights[i] else 0 for i in range(len(heights))])

File: 2022/07/06/heightChecker.ts
/**
 * https://leetcode.com/problems/height-checker/
 * @author Andrew Jarombek
 * @since 7/6/2022
 */

function heightChecker(heights: number[]): number {
    let orderedHeights = [...heights];
    orderedHeights.sort((x, y) => x - y);

    let result = 0;

    for (let i = 0; i < heights.length; i++) {
        result += heights[i] != orderedHeights[i] ? 1 : 0;
    }

    return result;
}